workflow/controller/operator.go

```diff
@@ -917,6 +917,7 @@ func (woc *wfOperationCtx) podReconciliation() error {
 
 			node.Message = "pod deleted"
 			node.Phase = wfv1.NodeError
+			node.Daemoned = nil
```

Comment:
what is this needed for please?

Reply:
When daemon pod deleted and `Daemoned` flag in node is true, there is a bug that the `Deaemoned` flag never removed and the whole workflow will never completed.